### PR TITLE
sdcard.py add support for multi-block read/write

### PR DESCRIPTION
**Pull request by peterhinch**
Tested with Sandisk Ultra 16GB micro SD cards. Not tested on a version 1 card as I don't posses one.

### CODE REVIEW

File: drivers/sdcard/sdcard.py [hunk 1]
```diff
@@ -176,29 +190,61 @@ def write(self, buf):
...
+
+            # send the data
+            self.write(TOKEN_DATA, buf)
+        else:
+            offset = 0
+            # CMD25: set write address for first block
+            if self.cmd(25, block_num * self.cdv, 0) != 0:
+                return 1
+            while nblocks:
+            # send the data
```
Comment (dpgeorge) [inline drivers/sdcard/sdcard.py]: Move comment 1 line up.

File: drivers/sdcard/sdcard.py [hunk 2]
```diff
@@ -176,29 +190,61 @@ def write(self, buf):
...
+                self.write(TOKEN_CMD25, buf[offset : offset+512])
```
Comment (dpgeorge) [inline drivers/sdcard/sdcard.py]: Don't want to make a copy of buf.  Best to use a memoryview.

**Comment by peterhinch** [comment]
Good point. Now done.

**Comment by dpgeorge** [comment]
Thanks!  I ran the test and it works for me.  A few cosmetic changes were made and then merged in 2bd758fe96e2913f48c1a0fdccafddc3dfb31cf9.

**Pull request closed without merging**